docs/features/kubernetes/authentication.md

```diff
@@ -0,0 +1,35 @@
+---
+id: kubernetes-auth
+title: Kubernetes Authentication
+description: Authentication in Kubernetes plugin
+---
+
+The authentication process in kubernetes is basically separate from backstage auth, the
+providers are configured so your kubernetes plugin can locate and access the clusters you
+have access to, the providers currently available are categorized in server side auth and
+client side auth.
+
+## Server Side Providers
+
+These providers authenticate your _application_ with the cluster, meaning anyone that is
+logged in into your backstage app will be granted the same access to Kubernetes objects.
+The providers available as server side are:
+
+- `aws`
+- `azure`
+- `googleServiceAccount`
+- `localKubectlProxy`
+- `serviceAccount`
+
+## Client Side Providers
+
+These providers authenticate your user with the cluster, each user will be requested for
+credentials and will have access to the clusters as long as the user has been authorized
+to access said cluster, if the cluster is listed in the `clusterLocatorMethods` in the
+config, but the user hasn't been authorized to access, the user will see the cluster
```

Comment:
Break up the run-on sentence
```suggestion
to access said cluster. If the cluster is listed in the `clusterLocatorMethods` in the
config, but the user hasn't been authorized to access, the user will see the cluster
```
Now there are other reasons Backstage will access a cluster besides having it listed in `clusterLocatorMethods` -- since there are [other cluster supplier implementations](https://backstage.io/docs/features/kubernetes/configuration#clusterlocatormethods) besides `config`, like `catalog`. @mclarke47 what do you think?

Reply:
Is the configuration for clusters in `catalog` similar to the one we use in `clusterLocatorMethods`? Is it scoped by catalog `user` or every user can see `catalog` clusters?

Reply:
I read again the docs and `catalog` is a type of `clusterLocatorMethods` that go inside the array, so I would say mentioning `clusterLocatorMethods` would cover `catalog` and the other methods.